Doesn't launch on cleanroom.

Says something about it being the wrong FML version or somethin, even though it's not.

```
[09:52:14] [main/FATAL]: A CRITICAL PROBLEM OCCURRED INITIALIZING MINECRAFT - LIKELY YOU HAVE AN INCORRECT VERSION FOR THIS FML
[09:52:14] [main/FATAL]: Unable to launch
java.lang.RuntimeException: kpan.light_and_shadow.util.ReflectionUtil$UnableToAccessFieldException: java.lang.NoSuchFieldException: transformers of class net.minecraft.launchwrapper.LaunchClassLoader
	at net.minecraftforge.fml.common.launcher.FMLDeobfTweaker.injectIntoClassLoader(FMLDeobfTweaker.java:74) ~[FMLDeobfTweaker.class:?]
	at top.outlands.foundation.LaunchHandler.launch(LaunchHandler.java:103) [LaunchHandler.class:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at top.outlands.foundation.boot.Foundation.main(Foundation.java:41) [foundation-0.14.0.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:105) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?]
Caused by: kpan.light_and_shadow.util.ReflectionUtil$UnableToAccessFieldException: java.lang.NoSuchFieldException: transformers of class net.minecraft.launchwrapper.LaunchClassLoader
	at kpan.light_and_shadow.util.ReflectionUtil.getPrivateFieldInternal(ReflectionUtil.java:151) ~[ReflectionUtil.class:?]
	at kpan.light_and_shadow.util.ReflectionUtil.getPrivateField(ReflectionUtil.java:76) ~[ReflectionUtil.class:?]
	at kpan.light_and_shadow.asm.core.AsmTransformer.rearrange(AsmTransformer.java:87) ~[AsmTransformer.class:?]
	at net.minecraftforge.fml.common.Loader.<init>(Loader.java:229) ~[Loader.class:?]
	at net.minecraftforge.fml.common.Loader.instance(Loader.java:191) ~[Loader.class:?]
	at net.minecraftforge.fml.common.launcher.FMLDeobfTweaker.injectIntoClassLoader(FMLDeobfTweaker.java:60) ~[FMLDeobfTweaker.class:?]
	... 7 more
Caused by: java.lang.NoSuchFieldException: transformers
	at java.lang.Class.getDeclaredField(Class.java:2782) ~[?:?]
	at kpan.light_and_shadow.util.ReflectionUtil.getPrivateFieldInternal(ReflectionUtil.java:147) ~[ReflectionUtil.class:?]
	at kpan.light_and_shadow.util.ReflectionUtil.getPrivateField(ReflectionUtil.java:76) ~[ReflectionUtil.class:?]
	at kpan.light_and_shadow.asm.core.AsmTransformer.rearrange(AsmTransformer.java:87) ~[AsmTransformer.class:?]
	at net.minecraftforge.fml.common.Loader.<init>(Loader.java:229) ~[Loader.class:?]
	at net.minecraftforge.fml.common.Loader.instance(Loader.java:191) ~[Loader.class:?]
	at net.minecraftforge.fml.common.launcher.FMLDeobfTweaker.injectIntoClassLoader(FMLDeobfTweaker.java:60) ~[FMLDeobfTweaker.class:?]
	... 7 more
Process exited with code 1.

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Change Log
 
+### [1.12.2-forge-1.2.0](https://github.com/KatatsumuriPan/LightAndShadow/releases/tag/1.12.2-forge-1.2.0) - 2025-01-24
+
+- Fix server crashing bug on CleanroomLoader.
+
 ### [1.12.2-forge-1.1.0](https://github.com/KatatsumuriPan/LightAndShadow/releases/tag/1.12.2-forge-1.1.0) - 2024-10-31
 
 - Fix server crashing bug.

--- a/gradle.properties
+++ b/gradle.properties
@@ -7,7 +7,7 @@ modGroup = kpan.light_and_shadow
 
 # Version of your mod.
 # This field can be left empty if you want your mod's version to be determined by the latest git tag instead.
-modVersion = 1.1.0
+modVersion = 1.2.0
 
 # Whether to use the old jar naming structure (modid-mcversion-version) instead of the new version (modid-version)
 includeMCVersionJar = false

--- a/src/main/java/kpan/light_and_shadow/asm/core/AsmPlugin.java
+++ b/src/main/java/kpan/light_and_shadow/asm/core/AsmPlugin.java
@@ -6,10 +6,12 @@ import kpan.light_and_shadow.ModTagsGenerated;
 import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;
 import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin.MCVersion;
 import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin.Name;
+import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin.SortingIndex;
 
 @IFMLLoadingPlugin.TransformerExclusions({ModTagsGenerated.MODGROUP + ".asm.core.", ModTagsGenerated.MODGROUP + ".asm.tf.", ModTagsGenerated.MODGROUP + ".util.ReflectionUtil"})
 @Name("AsmPlugin")
 @MCVersion("1.12.2")
+@SortingIndex(1357)
 public class AsmPlugin implements IFMLLoadingPlugin {
 
     public AsmPlugin() {

--- a/src/main/java/kpan/light_and_shadow/asm/core/AsmTransformer.java
+++ b/src/main/java/kpan/light_and_shadow/asm/core/AsmTransformer.java
@@ -1,20 +1,11 @@
 package kpan.light_and_shadow.asm.core;
 
-import java.util.List;
-import kpan.light_and_shadow.asm.core.adapters.InjectInstructionsAdapter;
-import kpan.light_and_shadow.asm.core.adapters.Instructions;
 import kpan.light_and_shadow.asm.core.adapters.MixinAccessorAdapter;
-import kpan.light_and_shadow.asm.core.adapters.MyClassVisitor;
 import kpan.light_and_shadow.asm.tf.TF_Entity;
-import kpan.light_and_shadow.util.ListUtil;
-import kpan.light_and_shadow.util.ReflectionUtil;
 import net.minecraft.launchwrapper.IClassTransformer;
-import net.minecraftforge.fml.common.asm.ASMTransformerWrapper.TransformerWrapper;
-import net.minecraftforge.fml.common.asm.transformers.DeobfuscationTransformer;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.ClassWriter;
-import org.objectweb.asm.MethodVisitor;
 
 public class AsmTransformer implements IClassTransformer {
 
@@ -40,7 +31,6 @@ public class AsmTransformer implements IClassTransformer {
             ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_FRAMES);// maxStack,maxLocal,frameの全てを計算
             // Adapterを通して書き換え出来るようにする。
             ClassVisitor cv = cw;
-            cv = rearrangeThisAfterDeobfuscation(cv, transformedName);
             cv = MixinAccessorAdapter.transformAccessor(cv, transformedName);
             cv = TF_Entity.appendVisitor(cv, transformedName);
 
@@ -60,47 +50,6 @@ public class AsmTransformer implements IClassTransformer {
             e.printStackTrace();
             throw e;
         }
-    }
-
-    private static ClassVisitor rearrangeThisAfterDeobfuscation(ClassVisitor cv, String className) {
-        if (!className.equals("net.minecraftforge.fml.common.Loader"))
-            return cv;
-        return new MyClassVisitor(cv, className) {
-            @Override
-            public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
-                MethodVisitor mv = super.visitMethod(access, name, desc, signature, exceptions);
-                if (name.equals("<init>")) {
-                    mv = InjectInstructionsAdapter.injectBeforeReturns(mv, name
-                            , Instructions.create()
-                                    .invokeStatic(AsmTransformer.class.getName(), "rearrange", AsmTypes.METHOD_VOID)
-                    );
-                    success();
-                }
-                return mv;
-            }
-        };
-    }
-
-    @SuppressWarnings("unused")
-    public static void rearrange() {
-        ClassLoader classLoader = AsmTransformer.class.getClassLoader();
-        List<IClassTransformer> transformers = ReflectionUtil.getPrivateField(classLoader, "transformers");
-
-        int thisIndex = ListUtil.indexOf(transformers, transformer -> {
-            if (transformer instanceof TransformerWrapper) {
-                IClassTransformer parent = ReflectionUtil.getPrivateField(TransformerWrapper.class, transformer, "parent");
-                return parent.getClass() == AsmTransformer.class;
-            }
-            return false;
-        });
-        int deobfIndex = ListUtil.indexOf(transformers, transformer -> {
-            if (transformer instanceof TransformerWrapper) {
-                IClassTransformer parent = ReflectionUtil.getPrivateField(TransformerWrapper.class, transformer, "parent");
-                return parent.getClass() == DeobfuscationTransformer.class;
-            }
-            return transformer.getClass().getName().equals("net.minecraftforge.fml.common.asm.transformers.DeobfuscationTransformer");
-        });
-        transformers.add(deobfIndex + 1, transformers.remove(thisIndex));
     }
 
 }

